force-app/main/default/classes/Email Recipes/InboundEmailHandlerRecipes.cls

```diff
@@ -0,0 +1,170 @@
+     * @param email    This is dependency injected by the system at runtime.
+     * @param envelope This is dependency injected by the system at runtime.
+     * @return         `Messaging.InboundEmailResult`
+     * @see FilesRecipes
+     */
+    public Messaging.InboundEmailResult handleInboundEmail(
+        Messaging.InboundEmail email,
+        Messaging.InboundEnvelope envelope
+    ) {
+        Messaging.InboundEmailResult result = new Messaging.InboundEmailresult();
+        String errorMessage = '';
+        try {
+            // identify contact
+            Contact sender = getContactBySender(email);
+            createFilesByEmailAttachments(email.binaryAttachments, sender.Id);
+            createEmailRecord(sender, email);
+            result.success = true;
+        } catch (InboundEmailHandlerRecipesException iehre) {
+            result.success = false;
+            errorMessage =
+                iehre.getMessage() +
+                ' ' +
+                iehre.getStackTraceString();
+        }
+
+        if (!result.success) {
+            result.message = errorMessage;
+        }
+
+        return result;
+    }
+
+    /**
+     * @description              This helper method bulk saves attachments from
+     * the incoming email. It relies on FilesRecipes.cls to do the actual
+     * creation of the Files attachments as well as publishing the file to the
+     * specified record.
+     * @param inboundAttachments
+     * @param contactId
+     */
+    private void createFilesByEmailAttachments(
+        List<Messaging.inboundEmail.BinaryAttachment> inboundAttachments,
+        Id contactId
+    ) {
+        // guard against invalid data
+        if (inboundAttachments == null || contactId == null) {
+            return;
+        }
+        List<FilesRecipes.FileAndLinkObject> toCreate = new List<FilesRecipes.FileAndLinkObject>();
+        for (
+            Messaging.inboundEmail.BinaryAttachment currentAttachment : inboundAttachments
+        ) {
+            FilesRecipes.FileAndLinkObject newFile = new FilesRecipes.FileAndLinkObject();
+            newFile.fileContents = currentAttachment.body;
+            newFile.fileName = currentAttachment.fileName;
+            newFile.attachedTo = contactId;
+            toCreate.add(newFile);
+        }
+        FilesRecipes.createFilesAttachedToRecords(toCreate);
+    }
+
+    /**
+     * @description         Determines if we have an existing contact record
+     * with an email address that matches the sender of this email.
+     * If we do not have a contact that matches, return a new contact object
+     * with the email address set.
+     * @param senderAddress
+     * @return              `Contact`
+     */
+    private Contact getContactBySender(Messaging.InboundEmail email) {
+        List<Contact> contactList = [
+            SELECT AccountId, Email
+            FROM Contact
+            WHERE Email = :email.fromAddress
+            WITH SECURITY_ENFORCED
```

Comment:
Yes and no. 

Class runs without sharing, and since this code is only-ever executed in a system context, the running user is effectively a Sys admin. In this case the `WITH SECURITY_ENFORCED` is effectively theater - it won't matter because of execution context and lack of sharing